src/main/scala/org/ergoplatform/modifiers/history/popow/PoPowHeader.scala

```diff
@@ -28,56 +38,115 @@ case class PoPowHeader(header: Header, interlinks: Seq[ModifierId]) extends Byte
 
   def height: Int = header.height
 
+  def checkProof(): Boolean = PoPowHeader.checkProof(interlinks, interlinksProof)
 }
 
 object PoPowHeader {
 
   import io.circe.syntax._
 
+  implicit val hf: HF = Blake2b256
+  val powScheme: AutolykosPowScheme = new AutolykosPowScheme(32, 26)
+  val nipopowAlgos: NipopowAlgos = new NipopowAlgos(powScheme)
+
+  def checkProof(interlinks: Seq[ModifierId], proof: BatchMerkleProof[Digest32]): Boolean = {
+    val fields = nipopowAlgos.packInterlinks(interlinks)
+    val tree = merkleTree(fields)
+    proof.valid(tree.rootHash)
+  }
+
   def fromBlock(b: ErgoFullBlock): Try[PoPowHeader] = {
+    val proof = nipopowAlgos.proofForInterlinkVector(b.extension).get
     NipopowAlgos.unpackInterlinks(b.extension.fields).map { interlinkVector =>
-      PoPowHeader(b.header, interlinkVector)
+      PoPowHeader(b.header, interlinkVector, proof)
     }
   }
 
   implicit val interlinksEncoder: Encoder[Seq[ModifierId]] = { interlinksVector: Seq[ModifierId] =>
     interlinksVector.map(id => id: String).asJson
   }
 
+  implicit val interlinkProofEncoder: Encoder[BatchMerkleProof[Digest32]] = { proof: BatchMerkleProof[Digest32] =>
```

Comment:
interlinkProofEncoder => batchMerkleProof (as it can be used not for interlinks vector only, but for e.g. proving inclusion of multiple transactions in a block)